src/DB_Command.php

```diff
@@ -904,34 +896,29 @@ public function search( $args, $assoc_args ) {
 					$outputted_table_column_once = false;
 					foreach ( $results as $result ) {
 						$col_val = $result->$column;
-						if ( preg_match_all( $search_regex, $col_val, $matches, $regex ? PREG_OFFSET_CAPTURE : PREG_PATTERN_ORDER ) ) {
+						if ( preg_match_all( $search_regex, $col_val, $matches, PREG_OFFSET_CAPTURE ) ) {
 							if ( ! $matches_only && ( ! $table_column_once || ! $outputted_table_column_once ) && ! $one_line ) {
 								WP_CLI::log( $table_column_val );
 								$outputted_table_column_once = true;
 							}
 							$pk_val = $primary_key ? ( $colors['id'][0] . $result->$primary_key . $colors['id'][1] . ':' ) : '';
 
 							$bits = array();
-							if ( $regex ) {
-								if ( null === $encoding ) {
-									$encoding = false;
-									if ( ( $before_context || $after_context ) && function_exists( 'mb_detect_encoding' ) ) {
-										$encoding = mb_detect_encoding( $col_val, null, true /*strict*/ );
-									}
-								}
-								foreach ( $matches[0] as $match_arr ) {
-									$match = $match_arr[0];
-									$offset = $match_arr[1];
-									// Offsets are in bytes, so need to use `strlen()` and  `substr()` before using `safe_substr()`.
-									$before = $before_context && $offset ? \cli\safe_substr( substr( $col_val, 0, $offset ), -$before_context, null /*length*/, false /*is_width*/, $encoding ) : '';
-									$after = $after_context ? \cli\safe_substr( substr( $col_val, $offset + strlen( $match ) ), 0, $after_context, false /*is_width*/, $encoding ) : '';
-									$bits[] = $before . $colors['match'][0] . $match . $colors['match'][1] . $after;
-								}
-							} else {
-								foreach ( $matches[0] as $key => $value ) {
-									$bits[] = $matches[1][ $key ] . $colors['match'][0] . $matches[2][ $key ] . $colors['match'][1] . $matches[3][ $key ];
+							$col_encoding = $encoding;
+							if ( null === $col_encoding ) {
+								$col_encoding = false;
+								if ( ( $before_context || $after_context ) && function_exists( 'mb_detect_encoding' ) ) {
+									$col_encoding = mb_detect_encoding( $col_val, null, true /*strict*/ );
 								}
 							}
+							foreach ( $matches[0] as $match_arr ) {
+								$match = $match_arr[0];
+								$offset = $match_arr[1];
+								// Offsets are in bytes, so need to use `strlen()` and `substr()` before using `safe_substr()`.
+								$before = $before_context && $offset ? \cli\safe_substr( substr( $col_val, 0, $offset ), -$before_context, null /*length*/, false /*is_width*/, $col_encoding ) : '';
+								$after = $after_context ? \cli\safe_substr( substr( $col_val, $offset + strlen( $match ) ), 0, $after_context, false /*is_width*/, $col_encoding ) : '';
```

Comment:
Just a general tip for code readability:

If you have scalar arguments where it is not immediately clear what their meaning is, you can use a simple assignment to make it clearer.

So, instead of `null /*length*/, false /*is_width*/`, you could also do `$length = null, $is_width = false`.

I personally find this easier to read.

Reply:
Yes, I agree, it is easier to read - however, gasp horror it's an unnecessary assignment! Knowingly doing an unnecessary assignment is the sort of thing that keeps me from sleeping! So I'll leave it for the mo...